### PR TITLE
RECIRC-106: Catch Http errors in recirculation data services

### PR DESCRIPTION
**Pull request by pauloslund**

### CODE REVIEW

File: extensions/wikia/Recirculation/services/DiscussionsDataService.class.php [hunk 1]
```diff
@@ -46,10 +46,17 @@ private function apiRequest() {
 		$endpoint = $wgCityId . '/forums/' . $wgCityId;
 
 		$url = $this->buildUrl( $endpoint, $options );
-		$data = Http::get( $url );
 
-		$obj = json_decode( $data, true );
-		return $obj['_embedded']['doc:threads'];
+		try {
+			$data = Http::get( $url );
```
Comment (gabrys) [inline extensions/wikia/Recirculation/services/DiscussionsDataService.class.php:51]: this doesn't throw on error, but rather return false

File: extensions/wikia/Recirculation/services/DiscussionsDataService.class.php [hunk 2]
```diff
@@ -46,10 +46,17 @@ private function apiRequest() {
...
+
+			$obj = json_decode( $data, true );
```
Comment (gabrys) [inline extensions/wikia/Recirculation/services/DiscussionsDataService.class.php:53]: this doesn't throw on error, but rather return null

File: extensions/wikia/Recirculation/services/DiscussionsDataService.class.php [hunk 3]
```diff
@@ -46,10 +46,17 @@ private function apiRequest() {
...
+			$posts = $obj['_embedded']['doc:threads'];
```
Comment (gabrys) [inline extensions/wikia/Recirculation/services/DiscussionsDataService.class.php:54]: this doesn't throw on error, but returns null too (plus emits a warning)

File: extensions/wikia/Recirculation/services/FandomDataService.class.php [hunk 4]
```diff
@@ -58,10 +58,16 @@ private function apiRequest( $type ) {
 		}
 
 		$url = $this->buildUrl( $endpoint, $options );
-		$data = ExternalHttp::get( $url );
 
-		$obj = json_decode( $data );
-		$posts = $this->dedupePosts( $obj->data );
+		try {
+			$data = ExternalHttp::get( $url );
```
Comment (gabrys) [inline extensions/wikia/Recirculation/services/FandomDataService.class.php:63]: same as above ;-)

File: extensions/wikia/Recirculation/services/DiscussionsDataService.class.php (hunk 1 above)
Comment (pauloslund) [inline extensions/wikia/Recirculation/services/DiscussionsDataService.class.php]: Generally yes, however it seems to throw exceptions when the route times out as per: https://wikia-inc.atlassian.net/browse/RECIRC-106. I could most like just wrap the `Http::get` in a try catch block, but seemed to make sense just to put it all in here even if the rest of the methods don't throw exceptions

**Pull request closed without merging**